clinical/dbaccess.py

```diff
@@ -13,112 +13,6 @@
 import psutil
 import subprocess
 
-def readconfig( config ):
-  """Reads parameters from a config file.
-
-  Args:
-    config (str): path to config file
-    If config does not exist the default will be used
-  
-  Returns:
-    dict: parameters from the config file (unparsed)
-  """
-  if os.path.isfile(config):
-    Configfile = config
-  else:
-    Configfile = "/home/hiseq.clinical/.scilifelabrc"
-  Params = {}
-  with open(Configfile, "r") as Confs:
-    for line in Confs:
-      if len(line) > 5 and not line[0] == "#":
-        line = line.rstrip()
-        pv = line.split(" ")
-        if len(pv) > 1:
-          arg = pv[0]
-          pv.pop(0)
-          Params[arg] = ' '.join(pv)
-        else:
-          Params[pv[0]] = pv[1]
-  return Params
-
-def create_tunnel(tunnel_cmd):
```

Comment:
This could be very nicely implemented as a context manager that handles everything from opening to closing the SSH connection. This makes for a very nice Pythonic API.

```
with create_tunnel([...]) as connection:
  # do something with the tunnel open

# the tunnel will be automatically closed here
```

Another advantage is that error handling can be contained in the function so we can make sure to close the tunnel no matter what happens in the code that runs inside the with-block!

More details: http://eigenhombre.com/2013/04/20/introduction-to-context-managers/